be/src/agent/heartbeat_server.cpp

```diff
@@ -190,6 +191,16 @@ StatusOr<HeartbeatServer::CmpResult> HeartbeatServer::compare_master_info(const
         }
     }
 
+    if (master_info.node_type == TNodeType::Backend && BackendOptions::is_cn()) {
```

Comment:
need to check if the node_type is set, in case the backend/compute node is upgraded to the newer version but the frontend is still running on the older version where there is no nodeType field in the thrift message.